Support "packed", bugfixes, add test cases

kamahen opened the pull request:
Add interoperability tests with C++, Python (not run as part of test_protobufs because they require protoc)
Bugfix: string(...)
Added: integer64(...) as an option for int32/int64 fields
Improved documentation
Added and reorganized unit tests

JanWielemaker commented:
I just did a quick read.   It is a lot of work and I'm barely familiar with protobufs.   As long as there is not something that breaks building and testing or introduces serious new dependencies, I'm happy.   That seems fine.

The only small note I have is that you replaced `` ```prolog `` blocks in the docs with `==`, I guess to try and fix your browser related fixed font issues?   The idea is to move in the direction of current markdown standards rather than away from it.

If you think it is fine, just merge into master and send me a note to update the submodule in the main repo.   I do keep that in my own hands, so I can at least verify that changes in submodules pass basic sanity checks and I'm roughly aware what is boiling where.

kamahen commented:
On Sun, 9 May 2021 at 02:02, Jan Wielemaker ***@***.***>
wrote:

> I just did a quick read. It is a lot of work and I'm barely familiar with
> protobufs. As long as there is not something that breaks building and
> testing or introduces serious new dependencies, I'm happy. That seems fine.
>

thanks!

> The only small note I have is that you replaced ```prolog blocks in the
> docs with ==, I guess to try and fix your browser related fixed font
> issues? The idea is to move in the direction of current markdown standards
> rather than away from it.
>
Earlier, I had noticed that some  *.md files didn't display properly on
github, so I had converted it to github markup ... when I realized that
some *.md files were used to create swi-prolog.org documentation, I
reverted the changes.

If you prefer ```prolog to ==, I can can re-revert those changes. ;)



> If you think it is fine, just merge into master and send me a note to
> update the submodule in the main repo. I do keep that in my own hands, so I
> can at least verify that changes in submodules pass basic sanity checks and
> I'm roughly aware what is boiling where.
>

I've sent a note to Jeff, in case he wants to review things. But if you're
planning a new release soon, I can merge into master any time and
incorporate Jeff's review (if any) later.

- p

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SWI-Prolog/contrib-protobufs/pull/4#issuecomment-835754142>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIIGNNJALKOVECLXTZ2DIWDTMZFRXANCNFSM44MCWEFA>
> .
>

JanWielemaker commented:
> If you prefer ```prolog to ==, I can can re-revert those changes. ;)

Using `` ``` `` (with our without `prolog` is to be preferred,  No reason for me to block merging though.

> But if you're planning a new release soon

I plan to release probably tomorrow.

    --- Jan
